### ISSUE TITLE
ButtonMenu a11y violation

### ISSUE THREAD

**Issue by nelsonchen90** [carbon-design-system/ibm-products#3519]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

> Describe in detail the issue you're having.

a11y violation when using PageHeader with `canary` ButtonMenu turned on and screen width is narrow so the buttons are collapsed down into a 'dropdown'

> Is this issue related to a specific component?

Yes, PagerHeader but I think ButtonMenu has the root cause

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Follow a11y guideline about "accessible label should contain or match the visible labels) or allow overrides

> What browser are you working in?

Chrome

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

1.59.0

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

IBM Knowledge Catalog / Cloudpak for data

## Steps to reproduce the issue

See the v1 storybook with ibm a11y checker:
![image](https://github.com/carbon-design-system/ibm-products/assets/10108576/d7e482cb-118a-4300-afc4-f6435143cb0c)

You can see this from v1 storybook
https://v1-ibm-products.carbondesignsystem.com/?path=/docs/ibm-products-components-page-header-pageheader--with-title#main-types

**Comment by ArslanYM** [comment]
Can I work on this?

**Comment by elycheea** [comment]
@ArslanYM If you’d like to contribute this, please feel free to open a PR to the `main_v1` branch.
